contrib_bots/lib/gcalendar.py

```diff
@@ -0,0 +1,233 @@
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        original_sender = message['sender_email']
+
+        content = message['content'].replace('@gcalendar ', '')
+
+        # Identify event name and date (ISO 8601 notation)
+        #   Expected formats:
+        #       "Meeting with John" 2003-01-14T13:37:00+01:00 2003-01-14T15:37:00+01:00
+        #       Trip 2003-01-14
+        input_re = '(\w+|"[\S ]+") ((\d{4}-(0[1-9]|1[1-2])-(0[1-9]|1[1-9]|2[1-9]|3[0-1])T(0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]:[0-5][0-9](\+|\-)(0[1-9]|1[1-4])\:[03]0 ?){2}|(\d{4}-(0[1-9]|1[1-2])-(0[1-9]|1[1-9]|2[1-9]|3[0-1])$))'
+
+        m = re.match(input_re, content)
+
+        if not m:
+            error = ('Unknown arguments set.\n'
+                'Usage examples:\n\n'
+                '```\n'
+                '@gcalendar "Meeting with John" 2003-01-14T13:37:00+01:00 2003-01-14T15:37:00+01:00\n'
+                '@gcalendar Trip 2003-01-14\n'
+                '```\n\n'
+                'Remember to use the [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) notation for dates: '
+                '`2020-04-16T13:58:13-02:00`')
+
+            reply_message(error, message, client)
+
+            return
+
+        title = m.group(1).strip('"')
+
+        splits = m.group(2).split();
+        start_str = splits[0]
+
+        event = {
+            'summary': title
+        }
+
+        if len(splits) != 2: # Full-day event
+            end_str = None
+            event.update({
+                'start': {
+                    'date': start_str
+                },
+                'end': {
+                    'date': start_str
+                }
+            })
+        else:
+            end_str = splits[1]
+            event.update({
+                'start': {
+                    'dateTime': start_str
+                },
+                'end': {
+                    'dateTime': end_str
+                }
+            })
+
+        credentials = get_credentials()
+        http = credentials.authorize(httplib2.Http())
+        service = discovery.build('calendar', 'v3', http=http)
+
+        try:
+            # TODO: Choose calendar ID from somwhere (see list_calendars note)
+            event = service.events().insert(calendarId='primary', body=event).execute()
+
+            if len(splits) != 2:
+                reply = (':calendar: Full day event created!\n'
+                    '> **{title}**, *on {startDate}*')
+            else:
+                reply = (':calendar: Event created!\n'
+                    '> **{title}**, *from {startDate} to {endDate}*')
+            
+            reply_message(reply.format(title=title, startDate=start_str, endDate=end_str), message, client)
+        except errors.HttpError as e:
+            error = (':warning: Error!\n'
+                '```\n'
+                '{err}\n'
+                '```\n')
+
+            reply_message(error.format(err=e), message, client)
```

Comment:
The user-friendly error implies parsing JSON, since the exception's message is a JSON string.

I had it already done, it's easy. But is it worth the library import and resources?